Source/deviceinfo/DeviceInfo.cpp

```diff
@@ -611,106 +614,574 @@ class DeviceInfoLink : public WPEFramework::RPC::SmartInterfaceType<WPEFramework
+    {
+        uint32_t result = Core::ERROR_NONE;
+        if(_modelYear.IsSet())
+        {
+            strncpy(buffer, _modelYear.Value().c_str(), _modelYear.Value().length());
+            *length = _modelYear.Value().length();
+        }
+        else
+        {
+            result = _deviceInfoLink.Deviceinfo_model_year(buffer,length);
+            if(result == Core::ERROR_NONE)
+            {
+                _modelYear = buffer;
+            }
+        }
+        return result;
+
+    }
+
+    uint32_t Deviceinfo_system_integrator_name(char buffer[], uint8_t* length)
+    {
+        int32_t result = Core::ERROR_NONE;
+        if(_systemIntegraterName.IsSet())
+        {
+            strncpy(buffer, _systemIntegraterName.Value().c_str(), _systemIntegraterName.Value().length());
+            *length = _systemIntegraterName.Value().length();
+        }
+        else
+        {
+            result = _deviceInfoLink.Deviceinfo_system_integrator_name(buffer,length);
+            if(result == Core::ERROR_NONE)
+            {
+                _systemIntegraterName = buffer;
+            }
+        }
+        return result;
+
+    }
+
+    uint32_t Deviceinfo_friendly_name(char buffer[], uint8_t* length)
+    {
+        int32_t result = Core::ERROR_NONE;
+        if(_friendlyName.IsSet())
+        {
+            strncpy(buffer, _friendlyName.Value().c_str(), _friendlyName.Value().length());
+            *length = _friendlyName.Value().length();
+        }
+        else
+        {
+            result = _deviceInfoLink.Deviceinfo_friendly_name(buffer,length);
+            if(result == Core::ERROR_NONE)
+            {
+                _friendlyName = buffer;
+            }
+        }
+        return result;
+
+    }
+
+    uint32_t Deviceinfo_platform_name(char buffer[], uint8_t* length)
+    {
+        int32_t result = Core::ERROR_NONE;
+        if(_platformName.IsSet())
+        {
+            strncpy(buffer, _platformName.Value().c_str(), _platformName.Value().length());
+            *length = _platformName.Value().length();
+        }
+        else
+        {
+            result = _deviceInfoLink.Deviceinfo_platform_name(buffer,length);
+            if(result == Core::ERROR_NONE)
+            {
+                _platformName = buffer;
+            }
+        }
+        return result;
+
+    }
+
+    uint32_t Deviceinfo_maximum_output_resolution(deviceinfo_output_resolution_t* value)
+    {
+        int32_t result = Core::ERROR_NONE;
+        if(_maxOutputResolution.IsSet())
+        {
+            *value = _maxOutputResolution;
+        }
+        else
+        {
+            result = _deviceInfoLink.Deviceinfo_maximum_output_resolution(value);
+            if(result == Core::ERROR_NONE)
+            {
+                _maxOutputResolution = *value;
+            }
+        }
+        return result;
+    }
+
+    uint32_t Deviceinfo_output_resolutions(deviceinfo_output_resolution_t value[], uint8_t* length)
+    {
```

Comment:
See comments at Deviceinfo_id, they most probably also reply here (and perhaps also at other functions below but will not repeat them)